queue/queue.go

```diff
@@ -81,10 +98,22 @@ func EventEnqueuer(ctx context.Context, q queue.Queue) lookout.EventHandler {
 			return err
 		}
 
+		if useCache {
+			cache.Store(e.ID(), true)
+			delayeCacheCleanup(cache, e.ID(), cleanupDelay)
+		}
+
 		return nil
 	}
 }
 
+func delayeCacheCleanup(cache *sync.Map, id lookout.EventID, cleanupDelay time.Duration) {
+	go func() {
```

Comment:
for a big number of events it would create lots of goroutines. Which isn't really a big problem (goroutines are cheap) but still better don't abuse this fact.

Standard approach for this task is to create one goroutine that cleans old values with some interval. In our case you can store timestamp instead of the `true` in map.

Example in one of the most popular cache lib in go: https://github.com/patrickmn/go-cache/blob/master/cache.go#L1071